crates/checkpoint-downloader/src/downloader.rs

```diff
@@ -226,11 +224,7 @@ impl ParallelCheckpointDownloaderInner {
                         .message_sender
                         .send(WorkerMessage::Shutdown)
                         .await?;
-                    let new_count = {
-                        let mut count = worker_count.write().await;
-                        *count -= 1;
-                        *count
-                    };
+                    let new_count = worker_count.fetch_sub(1, std::sync::atomic::Ordering::Relaxed);
```

Comment:
Is there a reason to use Relaxed here? Might as well use SeqCst, no? I don't see a memory fence being a problem here.

Reply:
I think Relaxed is just more performant (because no thread synchronization is needed) and the correctness isn't impacted from its usage.

Reply:
Good point, thanks.